Re-applying the form char length validation and conditional radio button

warrickct opened the pull request:
Adds a conditionally rendered radio button option that activates the 'New Project' storage request form steps.

Added a validator limit of 45 characters for storage folder short names.

warrickct committed:
Re-applying the form char length validation and conditional radio button
commits (reverting the revert)

This reverts commit 4ec5fc876d811d01b92a51311f6288c75ca2547e.

warrickct committed:
Added enforced character limit rather than the error field. Treat request JSON structure as if it is a new storage request.

warrickct committed:
Added value remapping for compatibility with cer automation scripts.

warrickct committed:
Testing what person data is retrieved

warrickct committed:
Adding PO email validator function.

warrickct committed:
Adjust PO email validator function.

warrickct committed:
debugging form validator errors.

warrickct committed:
Further debugging.

warrickct committed:
Updated error message slightly. PO validator working.

warrickct committed:
Removed logging.

warrickct committed:
Shortened error message.

warrickct committed:
Switched radio button to indented checkbox.

warrickct committed:
Added positive look before in email address validator.

devslashnoel commented on src/app/components/request-storage/request-storage.component.ts:
```diff
@@ -321,59 +372,64 @@ export class RequestStorageComponent implements OnInit, OnDestroy, CanComponentD
       email: new FormControl(person.email, [
         Validators.required,
-        Validators.pattern('.*(aucklanduni.ac.nz|auckland.ac.nz)$')
+        Validators.pattern(/.*(?<=\@|\.)(aucklanduni.ac.nz|auckland.ac.nz)$/),
```
Got a report saying the Hub's storage request feature isn't working in Safari. Investigated and found Safari does not support this regular expression. This error is given:
`SyntaxError: Invalid regular expression: invalid group specifier name` :thinking:

devslashnoel replied:
Looks like the regex lookbehind syntax isn't supported by Safari yet :disappointed: https://caniuse.com/js-regexp-lookbehind

devslashnoel committed:
Change storage member email regex to work in Safari
- Safari does not support lookbehind as of Jan 2021. https://caniuse.com/js-regexp-lookbehind
- Add some Emacs temp files to gitignore

devslashnoel commented:
Fixed Safari regex issue and deployed to dev, test and prod.

devslashnoel committed:
Minor fix for storage user add regex
- Email addresses like example@.auckland.ac.nz was being parsed as ok.
- Change regex to fix this.
